### PR TITLE
Class HelperTreeCategoriesCore is not translated

### LINKED ISSUE #20348: class HelperTreeCategoriesCore is not translated
Some text in this class is hardcoded, like "Expand All", "Collapse All", etc.

These strings should be translatable.

>Edited by Khouloud

Steps to reproduce the issue
1. Go to BO => Shop Parameters => General page 
2. Enable multishop context
3. Go to BO => Advanced Parameters => Multistore
4. Edit a shop
5. Change the language of the employee to any other language
`Collapse All`, `Expand All`, `Check All`, `Uncheck All` expressions are always displayed in English

### PR DESCRIPTION
**Pull request by PululuK**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop 
| Description?  | Some text in this class is hardcoded, like "Expand All", "Collapse All", etc.These strings should be translatable..
| Type?         | bug fix 
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes https://github.com/PrestaShop/PrestaShop/issues/20348
| How to test?  | See https://github.com/PrestaShop/PrestaShop/issues/20348 by @cortier

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/20656)

### CODE REVIEW

**Review by PierreRambaud — DISMISSED** [review]

**Review by atomiix — DISMISSED** [review]

File: classes/helper/HelperTreeCategories.php [hunk 1]
```diff
@@ -371,14 +371,14 @@ public function render($data = null)
         }
 
         $collapse_all = new TreeToolbarLink(
-            'Collapse All',
+            $this->translator->trans('Collapse All', [], 'Admin.Actions'),
```
Comment (LouiseBonnard) [inline classes/helper/HelperTreeCategories.php]:
```suggestion
            $this->translator->trans('Collapse all', [], 'Admin.Actions'),
```

File: classes/helper/HelperTreeCategories.php [hunk 2]
```diff
@@ -371,14 +371,14 @@ public function render($data = null)
...
             '#',
             '$(\'#' . $this->getId() . '\').tree(\'collapseAll\');$(\'#collapse-all-' . $this->getId() . '\').hide();$(\'#expand-all-' . $this->getId() . '\').show(); return false;',
             'icon-collapse-alt'
         );
         $collapse_all->setAttribute('id', 'collapse-all-' . $this->getId());
         $expand_all = new TreeToolbarLink(
-            'Expand All',
+            $this->translator->trans('Expand All', [], 'Admin.Actions'),
```
Comment (LouiseBonnard) [inline classes/helper/HelperTreeCategories.php]:
```suggestion
            $this->translator->trans('Expand all', [], 'Admin.Actions'),
```

File: classes/helper/HelperTreeCategories.php [hunk 3]
```diff
@@ -389,14 +389,14 @@ public function render($data = null)
 
         if ($this->useCheckBox()) {
             $check_all = new TreeToolbarLink(
-                'Check All',
+                $this->translator->trans('Check All', [], 'Admin.Actions'),
```
Comment (LouiseBonnard) [inline classes/helper/HelperTreeCategories.php]:
```suggestion
                $this->translator->trans('Check all', [], 'Admin.Actions'),
```

File: classes/helper/HelperTreeCategories.php [hunk 4]
```diff
@@ -389,14 +389,14 @@ public function render($data = null)
...
                 '#',
                 'checkAllAssociatedCategories($(\'#' . $this->getId() . '\')); return false;',
                 'icon-check-sign'
             );
             $check_all->setAttribute('id', 'check-all-' . $this->getId());
             $uncheck_all = new TreeToolbarLink(
-                'Uncheck All',
+                $this->translator->trans('Uncheck All', [], 'Admin.Actions'),
```
Comment (LouiseBonnard) [inline classes/helper/HelperTreeCategories.php]:
```suggestion
                $this->translator->trans('Uncheck all', [], 'Admin.Actions'),
```

File: classes/helper/HelperTreeShops.php [hunk 5]
```diff
@@ -124,25 +124,25 @@ public function render($data = null, $use_default_actions = true, $use_selected_
         if ($use_default_actions) {
             $this->setActions([
                 new TreeToolbarLink(
-                    'Collapse All',
+                    $this->translator->trans('Collapse All', [], 'Admin.Actions'),
```
Comment (LouiseBonnard) [inline classes/helper/HelperTreeShops.php]:
```suggestion
                    $this->translator->trans('Collapse all', [], 'Admin.Actions'),
```

File: classes/helper/HelperTreeShops.php [hunk 6]
```diff
@@ -124,25 +124,25 @@ public function render($data = null, $use_default_actions = true, $use_selected_
...
                     '#',
                     '$(\'#' . $this->getId() . '\').tree(\'collapseAll\'); return false;',
                     'icon-collapse-alt'
                 ),
                 new TreeToolbarLink(
-                    'Expand All',
+                    $this->translator->trans('Expand All', [], 'Admin.Actions'),
```
Comment (LouiseBonnard) [inline classes/helper/HelperTreeShops.php]:
```suggestion
                    $this->translator->trans('Expand all', [], 'Admin.Actions'),
```

File: classes/helper/HelperTreeShops.php [hunk 7]
```diff
@@ -124,25 +124,25 @@ public function render($data = null, $use_default_actions = true, $use_selected_
...
                     '#',
                     '$(\'#' . $this->getId() . '\').tree(\'expandAll\'); return false;',
                     'icon-expand-alt'
                 ),
                 new TreeToolbarLink(
-                    'Check All',
+                    $this->translator->trans('Check All', [], 'Admin.Actions'),
```
Comment (LouiseBonnard) [inline classes/helper/HelperTreeShops.php]:
```suggestion
                    $this->translator->trans('Check all', [], 'Admin.Actions'),
```

File: classes/helper/HelperTreeShops.php [hunk 8]
```diff
@@ -124,25 +124,25 @@ public function render($data = null, $use_default_actions = true, $use_selected_
...
                     '#',
                     'checkAllAssociatedShops($(\'#' . $this->getId() . '\')); return false;',
                     'icon-check-sign'
                 ),
                 new TreeToolbarLink(
-                    'Uncheck All',
+                    $this->translator->trans('Uncheck All', [], 'Admin.Actions'),
```
Comment (LouiseBonnard) [inline classes/helper/HelperTreeShops.php]:
```suggestion
                    $this->translator->trans('Uncheck all', [], 'Admin.Actions'),
```

**Review by LouiseBonnard — COMMENTED** [review]
Nice, thanks @PululuK, I take this opportunity to remove unnecessary capital letters in those wordings. ;-)

**Review by Progi1984 — APPROVED** [review]

**Review by NeOMakinG — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Review by LouiseBonnard — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by khouloudbelguith** [comment]
Hi @PululuKn

The `Collapse all`, `Expand all`, `Check all`, `Uncheck all` expressions are not translated.
1. **Collapse all**
image
2. **Expand all**
image
3. **Check all**
image
4. **Uncheck all**
image

**Before**
`Collapse All`, `Expand All`, `Check All`, `Uncheck All`
image

After
`Collapse all`, `Expand all`, `Check all`, `Uncheck all`
image


Thanks to check and feedback.

**Comment by PululuK** [comment]
Hi @khouloudbelguith  
Yes ! You try to translate `Check All` instead of `Check all`

@LouiseBonnard I think I should remove your suggestions, because on the translation file the words are in capital.

**Comment by khouloudbelguith** [comment]
@PululuK,

There are only those expressions
`Collapse All`, `Expand All`, `Check All`, `Uncheck All`
Those expressions: `Collapse all`, `Expand all`, `Check all`, `Uncheck all` do not exist.
I attached a screen record
https://www.awesomescreenshot.com/video/824040?key=819aefcc08d52435ed28f97895ca9bc1

Thanks!

**Comment by LouiseBonnard** [comment]
Hi @khouloudbelguith, @PululuK, it is the right behavior: removing capital letters will create new wordings, and all the 1.7.8 new wordings are not part of the translation files yet (but will be once the feature freeze declared), this is why it is not translated. :-)

**Comment by PululuK** [comment]
@LouiseBonnard  @khouloudbelguith  I have to do something ?

**Comment by LouiseBonnard** [comment]
Nope, thanks for asking @PululuK, those new wordings will be made translatable right before releasing this bug fix so there should not be any translation problem. ;-)

**Comment by khouloudbelguith** [comment]
@LouiseBonnard, thanks :slightly_smiling_face: 
@PululuK, as discussed with @LouiseBonnard it is ok :heavy_check_mark: 

Thanks!

**Comment by Progi1984** [comment]
Thanks @PululuK & @khouloudbelguith

**Pull request merged**